clippy_lints/src/casts/utils.rs

```diff
@@ -1,14 +1,55 @@
 use clippy_utils::ty::{EnumValue, read_explicit_enum_value};
-use rustc_middle::ty::{self, AdtDef, IntTy, Ty, TyCtxt, UintTy, VariantDiscr};
+use rustc_middle::ty::{self, AdtDef, FloatTy, IntTy, Ty, TyCtxt, UintTy, VariantDiscr};
 
-/// Returns the size in bits of an integral type, or `None` if `ty` is not an
-/// integral type.
-pub(super) fn int_ty_to_nbits(tcx: TyCtxt<'_>, ty: Ty<'_>) -> Option<u64> {
-    match ty.kind() {
-        ty::Int(IntTy::Isize) | ty::Uint(UintTy::Usize) => Some(tcx.data_layout.pointer_size.bits()),
-        ty::Int(i) => i.bit_width(),
-        ty::Uint(i) => i.bit_width(),
-        _ => None,
+/// Returns the size in bits of an integral type.
+/// Panics if the type is not an int or uint variant
+pub(super) fn int_ty_to_nbits(typ: Ty<'_>, tcx: TyCtxt<'_>) -> u64 {
+    match typ.kind() {
+        ty::Int(i) => match i {
+            IntTy::Isize => tcx.data_layout.pointer_size.bits(),
+            IntTy::I8 => 8,
+            IntTy::I16 => 16,
+            IntTy::I32 => 32,
+            IntTy::I64 => 64,
+            IntTy::I128 => 128,
+        },
+        ty::Uint(i) => match i {
+            UintTy::Usize => tcx.data_layout.pointer_size.bits(),
+            UintTy::U8 => 8,
+            UintTy::U16 => 16,
+            UintTy::U32 => 32,
+            UintTy::U64 => 64,
+            UintTy::U128 => 128,
+        },
+        _ => unreachable!(),
+    }
+}
```

Comment:
You are reverting (here and at other places) changes that were committed yesterday. Please don't do that. And please explain what your PR is doing, what it fixes, why it is necessary, and when you change existing things why your solution is better than the one in place. For example, why would be panicking best than returning `None` here? The `Option` version lets us check if this is an integral type at the same type, which is almost always necessary from the caller point of view anyway. Also, it is shorter by using `i.bit_width()` rather than hard-coding every case.

Reply:
According to your approach, how can we make use of int_ty_to_nbits, float_ty_to_nbits, and float_ty_to_mantissa_nbits at the same time? Should all three return Option, and then handle the None cases externally?

Reply:
I don't know what you're trying to achieve, so this is a difficult question to answer.